packages/connect/src/base/entry.ts

```diff
@@ -272,7 +288,9 @@ export class EntryNodes extends EventEmitter {
 
     const positiveOnes = results.findIndex((result: ConnectionResult) => result.entry.latency >= 0)
 
-    const previous = new Set<string>(this.usedRelays.map((ma) => relayFromRelayAddress(ma).toB58String()))
+    const previous = new Set<string>(
+      this.usedRelays.map((ma) => relayFromRelayAddress(ma.ourCircuitAddress).toB58String())
```

Comment:
Again might make more sense to use the helper function to get the circuit addresses.